src/utils/time.ts

```diff
@@ -26,10 +26,12 @@ export interface InternalTimeRange {
  */
 export const getValidTime = (date: TimeInput, timezone: string | undefined): number => {
   const tz = timezone || "system";
-  const dateInMillis = typeof date === "number" ? date : DateTime.fromISO(date, { zone: tz }).toMillis();
-  if (Number.isNaN(dateInMillis)) {
-    return new Date().getTime();
-  }
+  const dateInMillis =
+    typeof date === "number"
+      ? date
+      : typeof date === "string"
+      ? DateTime.fromISO(date, { zone: tz }).toMillis()
+      : DateTime.fromJSDate(date, { zone: tz }).toMillis();
 
   return dateInMillis;
```

Comment:
Rewrite this part with a switch case.
It will be useful in future in case we want to add some handling or log something.